src/Compilers/Core/Portable/Compilation/IOperation.cs

```diff
@@ -35,95 +44,164 @@ public interface IOperation
+        ConversionExpression = 0x1b,
+        /// <summary>Indicates an <see cref="IInvocationExpression"/>.</summary>
+        InvocationExpression = 0x1c,
+        /// <summary>Indicates an <see cref="IArrayElementReferenceExpression"/>.</summary>
+        ArrayElementReferenceExpression = 0x1d,
+        /// <summary>Indicates an <see cref="IPointerIndirectionReferenceExpression"/>.</summary>
+        PointerIndirectionReferenceExpression = 0x1e,
+        /// <summary>Indicates an <see cref="ILocalReferenceExpression"/>.</summary>
+        LocalReferenceExpression = 0x1f,
+        /// <summary>Indicates an <see cref="IParameterReferenceExpression"/>.</summary>
+        ParameterReferenceExpression = 0x20,
+        /// <summary>Indicates an <see cref="ISyntheticLocalReferenceExpression"/>.</summary>
+        SyntheticLocalReferenceExpression = 0x21,
+        /// <summary>Indicates an <see cref="IFieldReferenceExpression"/>.</summary>
+        FieldReferenceExpression = 0x22,
+        /// <summary>Indicates an <see cref="IMethodBindingExpression"/>.</summary>
+        MethodBindingExpression = 0x23,
+        /// <summary>Indicates an <see cref="IPropertyReferenceExpression"/>.</summary>
+        PropertyReferenceExpression = 0x24,
+        /// <summary>Indicates an <see cref="IEventReferenceExpression"/>.</summary>
+        EventReferenceExpression = 0x25,
+        /// <summary>Indicates an <see cref="ILateBoundMemberReferenceExpression"/>.</summary>
+        LateBoundMemberReferenceExpression = 0x26,
+        /// <summary>Indicates an <see cref="IUnaryOperatorExpression"/>.</summary>
+        UnaryOperatorExpression = 0x27,
+        /// <summary>Indicates an <see cref="IBinaryOperatorExpression"/>.</summary>
+        BinaryOperatorExpression = 0x28,
+        /// <summary>Indicates an <see cref="IConditionalChoiceExpression"/>.</summary>
+        ConditionalChoiceExpression = 0x29,
+        /// <summary>Indicates an <see cref="INullCoalescingExpression"/>.</summary>
+        NullCoalescingExpression = 0x2a,
+        /// <summary>Indicates an <see cref="ILambdaExpression"/>.</summary>
+        LambdaExpression = 0x2b,
+        /// <summary>Indicates an <see cref="IObjectCreationExpression"/>.</summary>
+        ObjectCreationExpression = 0x2c,
+        /// <summary>Indicates an <see cref="ITypeParameterObjectCreationExpression"/>.</summary>
+        TypeParameterObjectCreationExpression = 0x2d,
+        /// <summary>Indicates an <see cref="IArrayCreationExpression"/>.</summary>
+        ArrayCreationExpression = 0x2e,
+        /// <summary>Indicates an <see cref="IDefaultValueExpression"/>.</summary>
+        DefaultValueExpression = 0x2f,
+        /// <summary>Indicates an <see cref="IInstanceReferenceExpression"/>.</summary>
+        InstanceReferenceExpression = 0x30,
+        /// <summary>Indicates an <see cref="IIsExpression"/>.</summary>
+        IsExpression = 0x33,
+        // TypeOperationExpression = 0x34,
```

Comment:
Do you think it worth to change the value of `SizeOfExpression` (to 35) and shift the following values?